components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/tfm_impl/attestation_core.c

```diff
@@ -707,20 +707,38 @@ attest_add_security_lifecycle_claim(struct attest_token_ctx *token_ctx)
         if (res) {
             return PSA_ATTEST_ERR_GENERAL;
         }
-        security_lifecycle = (enum tfm_security_lifecycle_t)slc_value;
+        switch (slc_value) {
+            case TFM_SLC_UNKNOWN:
+                security_lifecycle = TFM_SLC_UNKNOWN;
+                break;
+            case TFM_SLC_ASSEMBLY_AND_TEST:
+                security_lifecycle = TFM_SLC_ASSEMBLY_AND_TEST;
+                break;
+            case TFM_SLC_PSA_ROT_PROVISIONING:
+                security_lifecycle = TFM_SLC_PSA_ROT_PROVISIONING;
+                break;
+            case TFM_SLC_SECURED:
+                security_lifecycle = TFM_SLC_SECURED;
+                break;
+            case TFM_SLC_NON_PSA_ROT_DEBUG:
+                security_lifecycle = TFM_SLC_NON_PSA_ROT_DEBUG;
+                break;
+            case TFM_SLC_RECOVERABLE_PSA_ROT_DEBUG:
+                security_lifecycle = TFM_SLC_RECOVERABLE_PSA_ROT_DEBUG;
+                break;
+            case TFM_SLC_DECOMMISSIONED:
+                security_lifecycle = TFM_SLC_DECOMMISSIONED;
+                break;
+            default:
+                security_lifecycle = TFM_SLC_UNKNOWN;
+        }
```

Comment:
Would this work without warnings?

```C
        switch (slc_value) {
            case TFM_SLC_UNKNOWN:
            case TFM_SLC_ASSEMBLY_AND_TEST:
            case TFM_SLC_PSA_ROT_PROVISIONING:
            case TFM_SLC_SECURED:
            case TFM_SLC_NON_PSA_ROT_DEBUG:
            case TFM_SLC_RECOVERABLE_PSA_ROT_DEBUG:
            case TFM_SLC_DECOMMISSIONED:
                security_lifecycle = slc_value;
            default:
                security_lifecycle = TFM_SLC_UNKNOWN;
        }
```

If not, what if `slc_value` were additionally changed to `int` and `get_int()` was used instead of `get_uint()`?